src/App.jsx

```diff
@@ -1,8 +1,33 @@
-import { useState } from "react";
+import { Routes, Route } from "react-router-dom";
+import { useEffect, useState } from "react";
 import "./App.css";
+import Header from "./components/Header";
+import Nav from "./components/Nav";
+import Signin from "./components/Signin";
+import Home from "./components/Home";
+import ArticleList from "./components/ArticleList";
+import { getCategories } from "./api";
 
 function App() {
-  return <></>;
+  const [category, setCategory] = useState([]);
+  useEffect(() => {
+    getCategories().then(({ data }) => {
+      setCategory(data);
+    });
+  }, []);
+  return (
+    <>
+      <Header />
+      <Routes>
+        <Route path="/" element={<Signin />} />
+        <Route
+          path="/home"
+          element={<Home category={category} setCategory={setCategory} />}
```

Comment:
Same with the Home component. Please get rid until you have it working 😄

Reply:
I'm actually using the Home component to generate my navbar but the Nav should probably be somewhere else as I need it on every page except the sign-in page